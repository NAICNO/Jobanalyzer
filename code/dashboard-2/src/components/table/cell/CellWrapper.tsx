import { ReactNode } from 'react'
import { Box, StyleProps } from '@chakra-ui/react'

interface CellWrapperProps {
  styleProps?: StyleProps
  children: ReactNode
}

export const CellWrapper = ({styleProps, children}: CellWrapperProps) => {
  return (
    <Box padding={1} {...styleProps}>
      {children}
    </Box>
  )
}

