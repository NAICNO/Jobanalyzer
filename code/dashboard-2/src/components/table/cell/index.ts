export { default as CellWithLink } from './CellWithLink'
export { default as CellWrapper } from './CellWrapper'
export { default as GenericCell } from './GenericCell'
export { default as GpuFieldCell } from './GpuFieldCell'
export { default as WorkingFieldCell } from './WorkingFieldCell'
