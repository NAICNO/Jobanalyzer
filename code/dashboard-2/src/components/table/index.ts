export * from './DasboardTable'
export * from './DashboardTableRow'
export * from './DeadWeightTable'
export * from './JobQueryResultsTable'
export * from './TableHeader'
export * from './TableRow'
export * from './ViolatingJobTable'
export * from './ViolatingUserTable'
