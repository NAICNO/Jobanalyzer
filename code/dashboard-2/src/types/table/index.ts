export * from './DashboardTableColumnHeader.ts'
export * from './DashboardTableItem.ts'
export * from './DeadWeightTableColumnHeader.ts'
export * from './DeadWeightTableItem.ts'
export * from './JobQueryJobId.ts'
export * from './JobQueryResultsTableColumnHeader.ts'
export * from './JobQueryResultsTableItem.ts'
export * from './ViolatingJobTableColumnHeader.ts'
export * from './ViolatingJobTableItem.ts'
export * from './ViolatingUserTableColumnHeader.ts'
export * from './ViolatingUserTableItem.ts'
