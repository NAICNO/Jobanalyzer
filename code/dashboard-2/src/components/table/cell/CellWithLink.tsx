import { Link as ReactRouterLink } from 'react-router-dom'
import { Link as ChakraLink } from '@chakra-ui/react'

import CellWrapper from './CellWrapper.tsx'

interface CellWithLinkProps {
  value: TextWithLink
}

const CellWithLink = ({value}: CellWithLinkProps) => {
  return (
    <CellWrapper styleProps={{paddingLeft: 4}}>
      <ChakraLink as={ReactRouterLink} to={value.link}>
        {value.text}
      </ChakraLink>
    </CellWrapper>
  )
}

export default CellWithLink
