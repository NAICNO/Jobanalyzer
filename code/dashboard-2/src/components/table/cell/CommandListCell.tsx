import { Text } from '@chakra-ui/react'

import { CellWrapper } from './CellWrapper.tsx'
import { breakText } from '../../../util'

interface CommandListCellProps {
  value: string;
}

export const CommandListCell = ({value}: CommandListCellProps) => {
  const brokenText = breakText(value)

  return (
    <CellWrapper>
      <Text>{brokenText}</Text>
    </CellWrapper>
  )
}
