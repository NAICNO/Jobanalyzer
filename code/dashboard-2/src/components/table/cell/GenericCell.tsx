import { Text } from '@chakra-ui/react'
import CellWrapper from './CellWrapper.tsx'

interface GenericCellProps {
  value: number;
}

const GenericCell = ({value}: GenericCellProps) => {
  return (
    <CellWrapper>
      <Text>{value}</Text>
    </CellWrapper>
  )
}

export default GenericCell
