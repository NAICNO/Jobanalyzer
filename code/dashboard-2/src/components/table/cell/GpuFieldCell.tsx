import { Text } from '@chakra-ui/react'

import { CELL_BACKGROUND_COLORS } from '../../../Constants.ts'
import { CellWrapper } from './CellWrapper.tsx'

interface GpuFieldCellProps {
  value: number;
}

export const GpuFieldCell = ({value}: GpuFieldCellProps) => {
  let backgroundColor = CELL_BACKGROUND_COLORS.NA
  if (value !== 0 || value === undefined) {
    backgroundColor = CELL_BACKGROUND_COLORS.DOWN
  }
  return (
    <CellWrapper styleProps={{backgroundColor}}>
      <Text>{value}</Text>
    </CellWrapper>
  )
}
