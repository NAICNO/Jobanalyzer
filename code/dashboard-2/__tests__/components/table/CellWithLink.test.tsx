import { render } from '@testing-library/react'
import { CellWithLink } from '../../../src/components/table/cell'
import { MemoryRouter } from 'react-router-dom'

const mockValue = {
  text: 'Test Link',
  link: '/test-link'
}

describe('CellWithLink', () => {
  it('renders correctly', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <CellWithLink value={mockValue}/>
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
