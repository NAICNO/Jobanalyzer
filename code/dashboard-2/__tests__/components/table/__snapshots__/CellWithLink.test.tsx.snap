// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CellWithLink > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-fclqwi"
  >
    <a
      class="chakra-link css-0"
      data-discover="true"
      href="/test-link"
    >
      Test Link
    </a>
  </div>
</DocumentFragment>
`;
